spyderlib/widgets/sourcecode/base.py

```diff
@@ -882,15 +895,17 @@ def hide_completion_widget(self):
     def show_completion_list(self, completions, completion_text="",
                              automatic=True):
         """Display the possible completions"""
-        if len(completions) == 0 or completions == [completion_text]:
+        c = completions
+        if len(c) == 0 or (len(c) == 1 and c[0][0] == completion_text):
             return
         self.completion_text = completion_text
         # Sorting completion list (entries starting with underscore are 
         # put at the end of the list):
-        underscore = set([comp for comp in completions
+        underscore = set([(comp, t) for (comp, t) in completions
                           if comp.startswith('_')])
-        completions = sorted(set(completions)-underscore, key=str_lower)+\
-                      sorted(underscore, key=str_lower)
+
+        completions = sorted(set(completions)-underscore, key=lambda x: str_lower(x[0]))+\
+                      sorted(underscore, key=lambda x: str_lower(x[0]))

```

Comment:
Could you break this line in two or more assignments? It's not easy to understand what it does :-)